### PR TITLE
Move the banner function from versions.jl to REPL

### PR DESCRIPTION
**Pull request by tecosaur**
The separate PR asked for in #51811.

### CODE REVIEW

**Review by staticfloat — APPROVED** [review]

File: base/version.jl [hunk 1]
```diff
@@ -267,72 +267,3 @@ end
 
 libllvm_path() = ccall(:jl_get_libllvm, Any, ())
```
Comment (LilithHafner) [inline base/version.jl:269]: The whitespace check is very particular about trailing newlines (there must be exactly one)
```suggestion
```

**Review by LilithHafner — COMMENTED** [review]
The other CI failures look bad but unrelated.

**Comment by LilithHafner** [comment]
Sorry this took forever to merge; I think nobody merged despite the merge me label because CI was all red because master's CI was all red.

**Comment by Drvi** [comment]
Hey folks, this PR broke `PackageCompiler` on master ([PR](https://github.com/JuliaLang/PackageCompiler.jl/pull/883) with a fix), which might be fair game since `banner` is not exported / public, but it makes me think that it would be good to test `PackageCompiler` as a part of Julias CI, since without functioning `PackageCompiler`, folks can't really test their apps against Julia nightly and help by reporting issues.

**Pull request merged**